### DISCUSSION TITLE
Option to pre-select one or more items in the Taxonomy Filter widget

### DISCUSSION THREAD

**Discussion by roboes** [elementor/elementor#25063] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

For now, it's not possible to pre-select one or more items from the Taxonomy Filter. By default, the "First Item" or all (i.e. no filter) all applied by default.

### Describe the solution you'd like.

Ideally there should be a option to pre-select one or more items that should be pre-selected once the page and Loop Grid is loaded. Then, the user can select other filters.

### Describe alternatives you've considered.

For now I am using a JavaScript solution using the WPCode plugin, as follows. Not an ideal solution, as the code is loaded after the page is loaded.

```.js
window.onload = function () {
	document.querySelector('button[data-filter="PRE_SELECTED_TAXONOMY"]').click();
}
```

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by Stelikas** [comment]
The above alternative works for desktop. 
Any alternatives in mobile version which is a dropdown selection?

**Reply by gnowland** [reply]
disable dropdown for mobile and use horizontal scrolling? 🤷‍♂️

**Comment by louiswol94** [comment]
If you direct users to the URL with the query params already added? It should filter it on page load if the query params are present in the URL.

**Reply by gnowland** [reply]
Not a solution when the loop grid is on the homepage :/  I hacked it with JS, below.

**Comment by gnowland** [comment]
I'm working around this with the following, but it's not ideal and can fail on slow networks (timeout compromise). Would be great if it was possible to just set a default on-load option in the editor!

```html
<script type="text/javascript">
const defaultTab = 'my-default-tag';
const loopGrid = document.getElementById('my-loop-grid');

if (!loopGrid.classList.contains('elementor-element-edit-mode')) {
   // Hide while loading to avoid flash of "all" items
    loopGrid.style.visibility = 'hidden';
}

// Set the selected element
function changeTab(value) {
    const filterElement = document.querySelector(`[data-filter="${value}"]`);
    if (filterElement) { /* && filterElement.getAttribute('aria-pressed') == 'false') { */
        filterElement.click();
        
        // Wait 1s and re-click if the initial click didn't stick
        if (filterElement.getAttribute('aria-pressed') == 'false') {
            setTimeout(function () {
                filterElement.click();
                console.warn('TaxonomyFilter: second click necessary')
            }, 100);
        }
        
        // Make items visible after a reasonable time necessary to execute the "filter" (4s)
        setTimeout(function () {
            loopGrid.style.visibility = 'visible';
        }, 400);

        // Scroll to filter element with an offset of 100px (delete this line if not needed)
        // window.scrollTo({ top: filterElement.getBoundingClientRect().top + window.scrollY - 100, behavior: 'smooth' });
    } else {
        console.error('filterElement does not exist yet')
    }
}

// Filter tabs based on URL query strings
// NOTE: adjust the timeout to suit your network speed (5ms)
window.addEventListener('load', function () {
    setTimeout(function () {
        const queryString = window.location.search;
        const urlParams = new URLSearchParams(queryString);
        
        // check through URL parameters to find if one starts with 'e-filter'
        if(urlParams.size > 0) {
            for (const [key, value] of urlParams) {
                if (key.startsWith('e-filter')) {
                    changeTab(value)
                    break;
                }
            }
        } else {
            // Set default tab if no query strings in URL
            changeTab(defaultTab)
        }
    }, 5);
});
</script>
```
